pkg/cmd/delete/environment.go

```diff
@@ -0,0 +1,160 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package delete
+
+import (
+	"os"
+
+	"github.com/litmuschaos/litmusctl/pkg/apis/environment"
+
+	"github.com/litmuschaos/litmusctl/pkg/apis"
+	"github.com/litmuschaos/litmusctl/pkg/utils"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+// experimentCmd represents the Chaos Experiment command
+var environmentCmd = &cobra.Command{
+	Use: "chaos-environment",
+	Short: `Delete a Chaos environment
+Example:
+#delete a Chaos Environment
+litmusctl delete chaos-environment c520650e-7cb6-474c-b0f0-4df07b2b025b --project-id=c520650e-7cb6-474c-b0f0-4df07b2b025b
+
+Note: The default location of the config file is $HOME/.litmusconfig, and can be overridden by a --config flag
+`,
+
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Fetch user credentials
+		credentials, err := utils.GetCredentials(cmd)
+		utils.PrintError(err)
+
+		projectID, err := cmd.Flags().GetString("project-id")
+		utils.PrintError(err)
+
+		environmentID, err := cmd.Flags().GetString("environment-id")
+		utils.PrintError(err)
+
+		// Handle blank input for project ID
+
+		if projectID == "" {
+			prompt := promptui.Prompt{
+				Label: "Enter the Project ID",
+			}
+			result, err := prompt.Run()
+			if err != nil {
+				utils.Red.Println("⛔ Error:", err)
+				os.Exit(1)
+			}
+			projectID = result
+		}
+
+		if environmentID == "" {
+			prompt := promptui.Prompt{
+				Label: "Enter the Environment ID",
+			}
+			result, err := prompt.Run()
+			if err != nil {
+				utils.Red.Println("⛔ Error:", err)
+				os.Exit(1)
+			}
+			environmentID = result
+		}
+
+		// Handle blank input for Chaos Environment ID
+		if environmentID == "" {
+			utils.Red.Println("⛔ Chaos Environment ID can't be empty!!")
+			os.Exit(1)
+		}
+
+		// Perform authorization
+		userDetails, err := apis.GetProjectDetails(credentials)
+		utils.PrintError(err)
+		var editAccess = false
+		var project apis.Project
+		for _, p := range userDetails.Data.Projects {
+			if p.ID == projectID {
+				project = p
+			}
+		}
+		for _, member := range project.Members {
+			if (member.UserID == userDetails.Data.ID) && (member.Role == "Owner" || member.Role == "Editor") {
+				editAccess = true
+			}
+		}
+		if !editAccess {
+			utils.Red.Println("⛔ User doesn't have edit access to the project!!")
+			os.Exit(1)
+		}
+
+		environmentList, err := environment.GetEnvironmentList(projectID, credentials)
+		if err != nil {
+			if strings.Contains(err.Error(), "permission_denied") {
+				utils.Red.Println("❌ You don't have enough permissions to access this resource.")
+				os.Exit(1)
+			} else {
+				utils.PrintError(err)
+				os.Exit(1)
+			}
+		}
+		environmentListData := environmentList.Data.ListEnvironmentDetails.Environments
+		for i := 0; i < len(environmentListData); i++ {
+			if environmentListData[i].EnvironmentID == environmentID {
+				if len(environmentListData[i].InfraIDs) > 0 {
```

Comment:
@Saranya-jena can you once confirm if we are deleting the infras from the infra array in environments collection on infra deletion

Reply:
yes we are!